src/webgl/interaction.js

```diff
@@ -1,107 +1,141 @@
     sensitivityY = sensitivityX;
   }
 
-  if (this.mouseIsPressed) {
-    var scaleFactor = this.height < this.width ? this.height : this.width;
-    var deltaTheta = -sensitivityX * (this.mouseX - this.pmouseX) / scaleFactor;
-    var deltaPhi = sensitivityY * (this.mouseY - this.pmouseY) / scaleFactor;
-
-    // camera position
-    var camX = this._renderer.cameraX;
-    var camY = this._renderer.cameraY;
-    var camZ = this._renderer.cameraZ;
-
-    // center coordinates
-    var centerX = this._renderer.cameraCenterX;
-    var centerY = this._renderer.cameraCenterY;
-    var centerZ = this._renderer.cameraCenterZ;
-
-    var diffX = camX - centerX;
-    var diffY = camY - centerY;
-    var diffZ = camZ - centerZ;
-
-    // get spherical coorinates for current camera position about origin
-    var camRadius = Math.sqrt(diffX * diffX + diffY * diffY + diffZ * diffZ);
-    // from https://github.com/mrdoob/three.js/blob/dev/src/math/Spherical.js#L72-L73
-    var camTheta = Math.atan2(diffX, diffZ); // equatorial angle
-    var camPhi = Math.acos(Math.max(-1, Math.min(1, diffY / camRadius))); // polar angle
-
-    // add mouse movements
-    camTheta += deltaTheta;
-    camPhi += deltaPhi;
-
-    // prevent rotation over the zenith / under bottom
-    if (camPhi > Math.PI) {
-      camPhi = Math.PI;
-    } else if (camPhi <= 0) {
-      camPhi = 0.001;
+  // default right-mouse and mouse-wheel behaviors (context menu and scrolling,
+  // respectively) are disabled here to allow use of those events for panning and
+  // zooming
+
+  // disable context menu for canvas element and add 'contextMenuDisabled'
+  // flag to p5 instance
+  if (this.contextMenuDisabled !== true) {
+    this.canvas.oncontextmenu = function() {
+      return false;
+    };
+    this._setProperty('contextMenuDisabled', true);
+  }
+
+  // disable default scrolling behavior on the canvas element and add
+  // 'wheelDefaultDisabled' flag to p5 instance
+  if (this.wheelDefaultDisabled !== true) {
+    this.canvas.onwheel = function() {
+      return false;
+    };
+    this._setProperty('wheelDefaultDisabled', true);
+  }
+
+  var scaleFactor = this.height < this.width ? this.height : this.width;
+
+  // ZOOM if there is a change in mouseWheelDelta
+  if (this._mouseWheelDeltaY !== this._pmouseWheelDeltaY) {
+    // zoom according to direction of mouseWheelDeltaY rather than value
+    if (this._mouseWheelDeltaY > 0) {
+      this._renderer._curCamera._orbit(0, 0, 0.5 * scaleFactor);
+    } else {
+      this._renderer._curCamera._orbit(0, 0, -0.5 * scaleFactor);
     }
+  }
+
+  if (this.mouseIsPressed) {
+    // ORBIT BEHAVIOR
+    if (this.mouseButton === this.LEFT) {
+      var deltaTheta =
+        -sensitivityX * (this.mouseX - this.pmouseX) / scaleFactor;
+      var deltaPhi = sensitivityY * (this.mouseY - this.pmouseY) / scaleFactor;
+      this._renderer._curCamera._orbit(deltaTheta, deltaPhi, 0);
+    } else if (this.mouseButton === this.RIGHT) {
```

Comment:
I tried this out in the manual-test-examples and the default interaction is very well tuned :)

maybe my lingo is out of date, but I might have expected movement along the y-axis to tilt the camera, and nothing to happen with the z-axis?

Reply:
oh sorry, I should have explained this better! I am looking at the manual-test-example for webgl/interaction. in the default orbitControl case, it looks like right click + x-axis movement pans but right click + y axis movement zooms in or out along the z-axis. is tilting not part of the default orbit control? I'm also not familiar with what the default expectation is for users navigating a 3D world—so feel free to ignore this if I'm out of my depth. ;)

Reply:
Ahh, ok!  Tilting is not a part of the default orbitControl as implemented here except that orbiting the center point tilts the camera to maintain the center point.  Panning behavior is also limited to the XZ plane.  This is how [three.js orbitControl](https://threejs.org/examples/?q=control#misc_controls_orbit) is implemented and I thought it was a good way to ensure that users don't 'get lost' by panning 'above' or 'below' their sketches.  

This arbitrary definition of a 'ground' as the XZ plane is definitely somewhat limiting if a user wants to move in the Y axis or has an especially 'tall' sketch, but I think is easier to use as a beginner.